Utils/check_protected_directories.py

```diff
@@ -0,0 +1,39 @@
+import argparse
+import re
+import sys
+
+PROTECTED_DIRECTORIES = [
+    ".circleci",
+    ".devcontainer",
+    ".github",
+    ".github/workflows",
+    ".gitlab",
+    ".guardrails",
+    ".hooks",
+    ".vscode",
+    "Templates",
+    "TestData",
+    "TestPlaybooks",
+    "Tests",
+    "Utils"
+]
+
+EXCEPTIONS = [
+    "Tests/conf.json"
+]
+
+
+def main(changed_files):
+    # Check if any protected directories have been modified
+    for changed_file in changed_files:
+        for protected_dir in PROTECTED_DIRECTORIES:
+            if re.match(f"^{protected_dir}/", changed_file) and not (changed_file in EXCEPTIONS):
```

Comment:
I'd suggest using `Path`'s `.parents` attribute, rather than a regex, i.e. `if Path(changed_file).parents[0] in PROTECTED_DIRECTORIES`, which saves the nested loop. 

(disclaimer: not sure if it's [0] or [-1])